lib/AutomaticEmails/WooCommerce/Events/PurchasedProduct.php

```diff
@@ -91,6 +91,7 @@ public function getProducts($productSearchQuery) {
     }
 
     $woocommerceProducts = array_map(function($product) {
+      assert($product instanceof \WP_Post);
```

Comment:
I think that type-hint would be better. Or do you think that we should have here to add `instanceof ` check? I don't know why I choose this wrong solution.

Reply:
I think we could go with type-hint. \WP_Post is a class from WordPress but it is very unlikely that it would change, because it would be a huge BC break.
And also I think that assert could be a good solution. The thing is just that its behavior is dependent on PHP configuration and since we don't control the environment that users use to run the plugin code, I think it would be better to use something else.